src/main/java/com/microsoft/aad/msal4j/TokenResponse.java

```diff
@@ -5,27 +5,29 @@
 
 import com.nimbusds.oauth2.sdk.ParseException;
 import com.nimbusds.oauth2.sdk.http.HTTPResponse;
-import com.nimbusds.oauth2.sdk.token.AccessToken;
-import com.nimbusds.oauth2.sdk.token.RefreshToken;
 import com.nimbusds.oauth2.sdk.util.JSONObjectUtils;
-import com.nimbusds.openid.connect.sdk.OIDCTokenResponse;
-import com.nimbusds.openid.connect.sdk.token.OIDCTokens;
 import lombok.AccessLevel;
 import lombok.Getter;
 import net.minidev.json.JSONObject;
 
 @Getter(AccessLevel.PACKAGE)
-class TokenResponse extends OIDCTokenResponse {
+class TokenResponse {
 
+    private String accessToken;
+    private String refreshToken;
+    private String idToken;
     private String scope;
     private String clientInfo;
     private long expiresIn;
     private long extExpiresIn;
     private String foci;
 
-    TokenResponse(final AccessToken accessToken, final RefreshToken refreshToken, final String idToken,
+    TokenResponse(final String accessToken, final String refreshToken, final String idToken,
                   final String scope, String clientInfo, long expiresIn, long extExpiresIn, String foci) {
-        super(new OIDCTokens(idToken, accessToken, refreshToken));
+
+        this.accessToken = accessToken;
+        this.refreshToken = refreshToken;
+        this.idToken = idToken; //TODO: Do we assume that IdToken won't be null or empty
```

Comment:
>TODO: Do we assume that IdToken won't be null or empty [](start = 34, length = 54)

I believe the cache currently depends on it. Not sure what ADFS and B2C does for all scenarios though?

Reply:
i do not think that we assume that, for client credential grant STS does not return idToken